### ISSUE TITLE
Proposal: V2 `course.yml` format

### ISSUE THREAD

**Issue by ejether** [FairwindsOps/reckoner#267]
The current format makes extensive use of dictionary yaml structures for the main sections `repositories` and `charts`
I propose a format change that make these list yaml structures instead, further, I propose an update to terminology, shifting `charts` -> `releases`

So a course:
```
namespace: kube-system
namespace_management:
  default:
    metadata:
      annotations:
        ManagedBy: com.fairwinds.reckoner
      labels:
        labelName: labelValue
    settings:
      overwrite: True
context: my-kube-cluster
repositories:
  stable:
    url: https://kubernetes-charts.storage.googleapis.com
hooks:
  pre_install:
    - ls ./
  post_install:
    - kubectl get namespaces
charts:
  my-metrics-server: # See below for chart schema
    repository: stable
    chart: metrics-server
 ```
 may, instead be structured
 ```
---
version: 2
namespace: kube-system
namespace_management:
  default:
    metadata:
      annotations:
        ManagedBy: com.fairwinds.reckoner
      labels:
        labelName: labelValue
    settings:
      overwrite: True
context: my-kube-cluster
repositories:
- name:  stable
   url: https://kubernetes-charts.storage.googleapis.com
hooks:
  pre_install:
    - ls ./
  post_install:
    - kubectl get namespaces
releases:
- name:my-metrics-server: # See below for chart schema
   repository: stable
   chart: metrics-server
 ```
 
 The reason for this proposed change is to:
* enable a better structure data format for future work (whispers in struct)
* allows more flexible deployment and use of helm3, (you can use the same release name across namespaces)
* more accurate terminology for what the reckoner manages. It does not manage charts after all, it manages releases

**Comment by sudermanjr** [comment]
Would this be a good time to convert `context` to a list as well, in order to support a single course.yaml to be applied to multiple contexts?

**Comment by ejether** [comment]
Maybe? Thats outside of how I think of `context` 🤔 

Would that, for example, be used to apply the same course.yml identically to multiple clusters as defined by the context? 
If so, that feels fraught and less declarative when you could start mix/matching using `-o` maybe?  I'd like to know more about what use case you're thinking?

Overall, I'm up for a discussion about structure and naming of _all_ of the headings though.

**Comment by sudermanjr** [comment]
It's an idea that got suggested/asked about at least once by a community member. Not something I feel super strongly about.

I definitely think this proposal is worth us all sitting dow and having a discussion about to layout exactly how we want to proceed. Overall, I think this is a great idea and definitely worth pursuing.

**Comment by sudermanjr** [comment]
We have decided that we won't make context a list.

**Comment by sudermanjr** [comment]
Fixed in #293.

Ultimately it actually worked out that the map works well. There's just a couple little changes in v2 as part of the golang re-write
